sdks/java/managed/src/main/java/org/apache/beam/sdk/managed/Managed.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.managed;
+
+import com.google.auto.value.AutoValue;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+
+public class Managed {
+  public static Read read() {
```

Comment:
I realize we don't have any sources yet, but what does the use look like here? 

`p.apply(Managed.read().???)`

Reply:
`p.apply(Managed.read().from(urn).withConfig(config))`

Reply:
I don't think the user will want to manually specify the urn, but rather something more user-friendly.

Reply:
So, we can add constants so that customers do not have to remember URNs (or SchemaTransform identifiers rather). So it will be actually in the form of,

`p.apply(Managed.read().for(Managed.ReadIds.ICEBERG).withConfig(config))`

Also changed `from` to `for` here which sounds more correct.

WDYT ?

Reply:
I suppose that's an option (though we'd have to do this per language). Is it bad, however, that a user could pass in a URN that's not managed and expect it to be managed? (I suppose this is a meta-question of who maintains this virtual list and what the actual contract is...)

Reply:
(though, for the first version, Ill be fine with a static list since we'll just have ICEBERG).

Reply:
In the interest of getting this PR in on time, I'm +1 on having a static list for now. 

This looks like a bigger design question about how best to maintain a list of supported identifiers that can be accessed from multiple SDKs, and whether we want this list living inside the repo or outside.

Reply:
Robert, are you OK with keeping this as a static list for the initial PR and making it dynamic (may be via codegen) in the future ?

I think a dynamic will list only be helpful if we can drop new jars with SchemaTransforms that we would like to manage to an older Beam version. At least initially, we will be very aware about the set of transforms we support (and the list will be small).

Reply:
Yes, I'm totally fine with a static list for the initial release. The key point is getting the API stable and generating the right protos.

Reply:
`p.apply(Managed.read(ManagedIO.ICEBERG).withConfig(config))`